rbac/management/inventory_replicator/inventory_replicator.py

```
@@ -0,0 +1,316 @@
+
+class InventoryReplicator(ABC):
+    """Type responsible for replicating relations to Kessel Relations."""
+
+    @abstractmethod
+    def replicate(self, event: ReplicationEvent):
+        """Replicate the given event to Kessel Relations."""
+        pass
+
+    def replicate_workspace(self, event: WorkspaceEvent, event_stream: WorkspaceEventStream):
+        """Replicate the given workspace event to Kessel Relations."""
+        pass
```

Ruff (0.16.1)

[warning] `InventoryReplicator.replicate_workspace` is an empty method in an abstract base class, but has no abstract decorator

(B027)